### PR TITLE
Trying to fix the error with docker in travisci

### PR DESCRIPTION
**Pull request by joaocc**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -9,8 +9,11 @@ go:
   - 1.5
 
 before_install:
-  - sudo apt-get update
-  - sudo apt-get install -o Dpkg::Options::="--force-confold" --force-yes -y docker-engine
+  - echo "Dpkg::Options { \"--force-confdef\"; \"--force-confold\"; }" |sudo tee -a /etc/apt/apt.conf.d/local
+  - sudo apt-get install apt-transport-https ca-certificates
+  - curl -sSL https://get.docker.com/ | sudo sh
+  - sudo pip2 install --upgrade pip setuptools wheel docker-compose
```
Comment (CWSpear) [inline .travis.yml]: The integration tests don't need `docker-compose`. We maybe could add some more tests to test the "run as a Docker container" version and use `docker-compose`, but for now, this isn't needed.

File: .travis.yml [hunk 2]
```diff
@@ -9,8 +9,11 @@ go:
   - 1.5
 
 before_install:
-  - sudo apt-get update
-  - sudo apt-get install -o Dpkg::Options::="--force-confold" --force-yes -y docker-engine
+  - echo "Dpkg::Options { \"--force-confdef\"; \"--force-confold\"; }" |sudo tee -a /etc/apt/apt.conf.d/local
```
Comment (CWSpear) [inline .travis.yml]: Call me neurotic, but I think there should be a space on both sides of a pipe (`|`).

**Comment by CWSpear** [comment]
So we'll see when Travis gets back, but I don't know if this will actually fix it. Travis and Docker is _incredibly_ finicky. I've tried looking at a few alternatives, but haven't really found anything.

**Comment by joaocc** [comment]
Not fixed it.
I think (very high level of guesswork here) that it may have to do with changes to Docker's API in the latest versions (hints here https://github.com/ContainX/docker-volume-netshare/issues/13).
J

**Comment by CWSpear** [comment]
These issues aren't related to how Docker is installed.

Go doesn't have a 1st party versioning for dependencies. They feel like they don't need it or whatever, and it's the dumbest thing ever. But one of the dependencies of this plugin keeps introducing breaking changes and so you can't build the binary. That's what's happening here.

So your PR is irrelevant, as it's trying to fix something that's not broken.

**Comment by CWSpear** [comment]
By the way, this was fixed in https://github.com/CWSpear/local-persist/commit/d5af6478ccfa3b74c24fdbf43abdbeb6e0b5b011

**Pull request closed without merging**